### PR TITLE
Update databricks-configs.md for merge settings

### PR DESCRIPTION
**Pull request by benc-db**
## What are you changing in this pull request and why?

1.9.1 dbt-databricks added a missed merge option, where you can update rows on non-match.  Updating the docs to cover it.

## Checklist
- [x] I have reviewed the [Content style guide](https://github.com/dbt-labs/docs.getdbt.com/blob/current/contributing/content-style-guide.md) so my content adheres to these guidelines.
- [x] The topic I'm writing about is for specific dbt version(s) and I have versioned it according to the [version a whole page](https://github.com/dbt-labs/docs.getdbt.com/blob/current/contributing/single-sourcing-content.md#adding-a-new-version) and/or [version a block of content](https://github.com/dbt-labs/docs.getdbt.com/blob/current/contributing/single-sourcing-content.md#versioning-blocks-of-content) guidelines.

### CODE REVIEW

File: website/docs/reference/resource-configs/databricks-configs.md [hunk 1]
```diff
@@ -437,8 +437,8 @@ Beginning with 1.9, `merge` behavior can be modified with the following addition
...
 - `not_matched_condition`: Condition to apply to the `WHEN NOT MATCHED [BY TARGET]` clause.  This condition further restricts the set of rows in the target that do not match the source that will be inserted into the merged table.
-- `not_matched_by_source_condition`: Condition to apply to the further filter `WHEN NOT MATCHED BY SOURCE` clause.  Only used in conjunction with `not_matched_by_source_action: delete`.
-- `not_matched_by_source_action`: If set to `delete`, a `DELETE` clause is added to the merge statement for `WHEN NOT MATCHED BY SOURCE`.
+- `not_matched_by_source_condition`: Condition to apply to the further filter `WHEN NOT MATCHED BY SOURCE` clause.  Only used in conjunction with `not_matched_by_source_action`.
+- `not_matched_by_source_action`: Action to apply when the condition is met.  For example: `update set t.attr1 = 'deleted', t.tech_change_ts = current_timestamp()`.
```
Comment (amychen1776) [inline website/docs/reference/resource-configs/databricks-configs.md]: @benc-db This wording is unclear to me. Is this a boolean field or one that takes inputs like delete (and maybe others?)

Comment (benc-db) [reply website/docs/reference/resource-configs/databricks-configs.md]: It takes an expression, like the example.

**Review by matthewshaver — APPROVED** [review]

File: website/docs/reference/resource-configs/databricks-configs.md [hunk 2]
```diff
@@ -437,8 +437,8 @@ Beginning with 1.9, `merge` behavior can be modified with the following addition
...
 - `not_matched_condition`: Condition to apply to the `WHEN NOT MATCHED [BY TARGET]` clause.  This condition further restricts the set of rows in the target that do not match the source that will be inserted into the merged table.
-- `not_matched_by_source_condition`: Condition to apply to the further filter `WHEN NOT MATCHED BY SOURCE` clause.  Only used in conjunction with `not_matched_by_source_action: delete`.
-- `not_matched_by_source_action`: If set to `delete`, a `DELETE` clause is added to the merge statement for `WHEN NOT MATCHED BY SOURCE`.
+- `not_matched_by_source_condition`: Condition to apply to the further filter `WHEN NOT MATCHED BY SOURCE` clause.  Only used in conjunction with `not_matched_by_source_action`.
+- `not_matched_by_source_action`: Action to apply when the condition is met.  For example: `not_matched_by_source_action: "update set t.attr1 = 'deleted', t.tech_change_ts = current_timestamp()"`.
```
Comment (matthewshaver) [inline website/docs/reference/resource-configs/databricks-configs.md]:
```suggestion
- `not_matched_by_source_action`: The action to apply when the condition is met. Configure as an expression. For example: `not_matched_by_source_action: "update set t.attr1 = 'deleted', t.tech_change_ts = current_timestamp()"`.
```

**Comment by matthewshaver** [comment]
@amychen1776 I've approved the editorial review but wanted to ensure your query was satisfied before merging.

**Review by amychen1776 — APPROVED** [review]

**Comment by mirnawong1** [comment]
looks like @amychen1776 has approved this so going to merge this pr. thanks @amychen1776 @matthewshaver and @benc-db !

**Pull request merged**